Enhance 404 message styling

Infinite-Null opened the pull request:
Closes: #69174

## What?
This PR improves the error message displayed on 404 (Not Found) pages in the Site Editor.

## Testing Instructions

1. Navigate to the Site Editor
2. Try to access a non-existent template by entering an invalid URL: 
```
/wp-admin/site-editor.php?p=%2Ftemplate-fdgfdgdfg
```
3. Verify the error appears as a proper error Notice component.

## Screenshots

### Desktop 

|Before|After|
|-|-|
|![image](https://github.com/user-attachments/assets/b94e296c-11fc-4155-b15a-89ad6229d7c9)|![image](https://github.com/user-attachments/assets/88598091-f6b9-4f15-b83e-30f1de728bc4)|

### Mobile 

|Before|After|
|-|-|
|![image](https://github.com/user-attachments/assets/c42e16de-7454-4f4a-9c26-eda37c9faad1)|![image](https://github.com/user-attachments/assets/dc348039-9fc9-43be-8b46-ed222daacb25)|

Infinite-Null committed:
Enhance 404 message styling with Notice component

Infinite-Null committed:
Improve 404 error message

Infinite-Null commented on packages/edit-site/src/components/layout/style.scss:
```diff
@@ -264,6 +264,7 @@ html.canvas-mode-edit-transition::view-transition-group(toggle) {
 
 .edit-site-layout__area__404 {
 	margin: $canvas-padding;
+	color: $gray-800;
```
After PR #69226 gets merged. This line won't be necessary.

Infinite-Null commented:
Hi @t-hamano, @carolinan and @joedolson,

I have included the Notice component in the 404 message in this PR. I would truly appreciate it if you could take a moment to review it and share your feedback.

t-hamano reviewed (commented):
Thanks for the PR!

Now that #69226 has been merged, the color overrides are no longer necessary. Can you rebase this PR?

Here are three things I noticed:

---

The browser console gives me an error: I think we need to replace the description wrapper with a `div` tag instead of a `p` tag:

```
react-dom.js?ver=18:73 Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
```

---

This may be a matter of preference, but you can also do something simpler like this:

```jsx
function NotFonundError() {
	return (
		<Notice status="error" isDismissible={ false }>
			{ __(
				'The requested page could not be found. Please check the URL.'
			) }
		</Notice>
	);
}

export const notFoundRoute = {
	areas: {
		mobile: (
			<SidebarNavigationScreenMain
				customDescription={ <NotFonundError /> }
			/>
		),
		content: (
			<Spacer padding={ 2 }>
				<NotFonundError />
			</Spacer>
		),
	},
};
```

---

I'm a bit undecided whether to use the error or warning status.

joedolson commented:
> I'm a bit undecided whether to use the error or warning status.

I'm somewhat inclined towards error; but also fairly undecided.

Infinite-Null committed:
Merge branch 'trunk' into feat/enhance-404-page-message

Infinite-Null committed:
Remove color style from `.edit-site-layout__area__404`

Infinite-Null committed:
Update 404 notice implementation with reusable component

Infinite-Null commented:
Hi @t-hamano,
Thank you for the feedback.
I have updated the implementation and made `NotFoundError ` a reusable component and used it in both content and mobile areas. Also removed:

```
.edit-site-layout__area__404 {
	margin: $canvas-padding;
}
```

as it is no longer required.

Please review it at your convenience.

t-hamano reviewed (commented):
Thanks for the update!

Can you fix this browser console error?

> The browser console gives me an error: I think we need to replace the description wrapper with a `div` tag instead of a `p` tag:
> 
> ```
> react-dom.js?ver=18:73 Warning: validateDOMNesting(...): <div> cannot appear as a descendant of <p>.
> ```

![image](https://github.com/user-attachments/assets/e3ccdd38-ec8b-4df1-963f-c969ae9e90d1)

Infinite-Null committed:
Fixed DOM nesting validation error by replacing p with div

Infinite-Null commented:
Hi @t-hamano,
I have fixed the console error as you suggested Please review it at your convenience.
Thank You!

t-hamano reviewed (approved):
I think the `error` status is fine for now.

By the way, we plan to display some text when you access a screen where the classic theme is not allowed to be used (#69005).

Let's take into consideration the status of the PR and if we think a `warning` status would be more appropriate, we can address it in a follow-up.

Infinite-Null commented:
Hi @t-hamano, I was wondering if adding an E2E test for the 404 page would be beneficial. I’m unsure of it. Do you think it’s worth adding?

t-hamano commented:
> I was wondering if adding an E2E test for the 404 page would be beneficial.

Yes, I think that's a good idea.
